File: Brute Force/acmicpc.net/15663/Main.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileInputStream;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.*;

class Main {
  static final String inputFilePath = "./# ing/sample.txt";
  static boolean[] c = new boolean[10];
  static int[] a = new int[10];
  static StringBuilder sb = new StringBuilder();
  static ArrayList<Integer> al = new ArrayList<Integer>();
  static int n, m;

  public static void main(String[] args) throws Exception {
    System.setIn(new FileInputStream(inputFilePath));
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    StringTokenizer st = new StringTokenizer(br.readLine());
    n = Integer.parseInt(st.nextToken());
    m = Integer.parseInt(st.nextToken());
    st = new StringTokenizer(br.readLine());
    for (int i = 0; i < n; i++) {
      int tmp = Integer.parseInt(st.nextToken());
      al.add(tmp);
    }
    Collections.sort(al);
    go(0, 0);
    st = new StringTokenizer(sb.toString(), "\\");
    ArrayList<String> strAl = new ArrayList<String>();
    while (st.hasMoreTokens()) {
      String line = st.nextToken();
      strAl.add(line);
    }
    ArrayList<String> ansAl = new ArrayList<String>();
    for (int i = 0; i < strAl.size(); i++) {
      if (!ansAl.contains(strAl.get(i))) {
        ansAl.add(strAl.get(i));
      }
    }
    for (int i = 0; i < ansAl.size(); i++) {
      System.out.println(ansAl.get(i));
    }
  }

  static void go(int index, int start) {
    if (index == m) {
      for (int i = 0; i < m; i++) {
        sb.append(a[i]);

        // System.out.print(a[i]);
        if (i < m - 1) {
          sb.append(" ");
          // System.out.print(" ");
        }
      }
      sb.append("\\");
      // System.out.println();
      return;
    }
    for (int i = 0; i < al.size(); i++) {
      if (c[i]) {
        continue;
      }
      c[i] = true;
      a[index] = al.get(i);
      go(index + 1, start);
      c[i] = false;
    }
  }
}

File: Brute Force/acmicpc.net/15663/Main.js
const inputFilePath = "/sample.txt";
let input = require("fs").readFileSync(__dirname + inputFilePath);
// let input = require("fs").readFileSync('/dev/stdin');
input = input.toString().trim().split("\n");
let firstLine = input[0].split(" ");
const N = Number(firstLine[0]);
const M = Number(firstLine[1]);
let array = input[1].split(" ").map(x => Number(x)).sort((a, b) => a - b);
array = calc(array);
let a = [];
let c = array[1];
array = array[0];
let ans = [];
go(0);
console.log(ans.join("\n").trim());

function calc(arr) {
  let a = [], b = [], prev;
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] !== prev) {
      a.push(arr[i]);
      b.push(1);
    } else {
      b[b.length - 1]++;
    }
    prev = arr[i];
  }
  return [a, b];
}

function go(index) {
  if (index === M) {
    ans.push(a.join(" "));
    return;
  }
  for (let i = 0; i < array.length; i++) {
    if (c[i] == 0) continue;
    c[i]--;
    a[index] = array[i];
    go(index + 1);
    c[i]++;
  }
}

